test/flags.go

```diff
@@ -13,9 +13,15 @@ var Flags = initializeFlags()
 
 // FlagsStruct is struct that defines testing options
 type FlagsStruct struct {
-	Kubeconfigs   string // Path to .kube/config
-	CatalogSource string // CatalogSource in the openshift-marketplace namespace for the serverless-operator Subscription
-	Channel       string // serverless-operator Subscription channel
+	Kubeconfigs     string // Path to .kube/config
+	CatalogSource   string // CatalogSource in the openshift-marketplace namespace for the Serverless operator Subscription
+	Channel         string // Serverless operator Subscription channel
+	Subscription    string // Serverless operator Subscription name
+	UpgradeChannel  string // Target OLM channel for upgrades
+	CSV             string // Target CSV for upgrades
+	ServingVersion  string // Target Serving version for upgrades
+	EventingVersion string // Target Eventing version for upgrades
+	OpenShiftImage  string // Target OpenShift image for upgrades
```

Comment:
It would be much easier to use environmental variables instead of go flags for those. Using https://github.com/kelseyhightower/envconfig

Reply:
Thanks for the suggestion. It's not clear to me how it would things easier though. We need to set the name, set the description, and the default value anyway. What would it actually bring? Curious...
The problem with kelseyhightoewer library is that there's no clear connection between environment variables and the fields of the given struct. Basically, you need to know that the given project is using this library otherwise it's confusing and hard to find where the fields are being set. You can't grep the code. There's no direct assignment.
OTOH, passing them as flags makes it clear where they're coming from.
When it comes to calling the tests, whether you pass the argument as a flag or you define an environment variable, it's more or less the same effort.
I used flags to make it consistent with the other flags that we already have such as `channel`, `catalogsource`. 
It's true that the upgrade tests still use two env variables: SKIP_SERVING_PREUPGRADE and UPGRADE_CLUSTER. However, I see them as something that is orthogonal - you don't have to use them - they're optional. Anyway. I could turn them into flags as well if this is desired.

Reply:
Imo, let's stay consistent and use flags.